### ISSUE TITLE
'super.visitNamedDeclaration()' is not invoked in the short-circuit case

### ISSUE THREAD

**Issue by pcl** [arturbosch/detekt#1014]
https://github.com/arturbosch/detekt/blob/3fb48aeaccb1024fc07e8a27cb49eeb3ec5a5bf9/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRules.kt#L57

**Comment by schalkms** [comment]
Thanks for the code-review and the heads-up.
Have you found any case where the `NamingRules` implementation does not work as expected? I couldn't come up with a test case.

**Comment by pcl** [comment]
I haven't, and I'd be fairly surprised if it's causing issues within the codebase currently.

I could imagine that if someone subclassed `NamingRules` and expected a certain behavior, it could turn into an issue. I brought it up just because it seemed different than other usages. 

Interestingly, this is a great use case for a custom detekt rule -- "guarantee that all methods named `visit*` in a `DetektVisitor` subclass call `super.visit*()` in all non-exception branches".

**Comment by schalkms** [comment]
> I could imagine that if someone subclassed NamingRules and expected a certain behavior, it could turn into an issue.

Agreed, but only if the sub class overrides `visit*()` and calls the base method.

> Interestingly, this is a great use case for a custom detekt rule -- "guarantee that all methods named visit* in a DetektVisitor subclass call super.visit*() in all non-exception branches".

This could be custom rule only for *detekt*, but I doubt that this would work for all visitor use cases.

**Comment by pcl** [comment]
For sure. Definitely a detekt-only sort of thing.
